src/modules/musig/main_impl.h

```diff
@@ -150,286 +160,303 @@ int secp256k1_musig_pubkey_tweak_add(const secp256k1_context* ctx, secp256k1_mus
-        secp256k1_gej_add_ge_var(&combined_noncej, &combined_noncej, &noncep, NULL);
-    }
-    secp256k1_sha256_finalize(&sha, nonce_commitments_hash);
-    /* If the signers' commitments changed between get_public_nonce and now we
-     * have to abort because in that case they may have seen our nonce before
-     * creating their commitment. That can happen if the signer_data given to
-     * this function is different to the signer_data given to get_public_nonce.
-     * */
-    if (session->has_secret_data
-            && memcmp(session->nonce_commitments_hash, nonce_commitments_hash, 32) != 0) {
+        secp256k1_sha256_write(&sha, buf, 32);
+    }
+    secp256k1_sha256_write(&sha, combined_pk32, 32);
+    secp256k1_sha256_write(&sha, msg, 32);
+    secp256k1_sha256_finalize(&sha, noncehash);
+    return 1;
+}
+
+static int secp256k1_musig_process_nonces_internal(const secp256k1_ecmult_context* ecmult_ctx, secp256k1_ge *combined_nonce, unsigned char *noncehash, secp256k1_gej *summed_noncesj, const unsigned char *combined_pk32, const unsigned char *msg) {
+    secp256k1_scalar b;
+    secp256k1_gej combined_noncej;
+    secp256k1_ge summed_nonces[2];
+
+    secp256k1_ge_set_gej(&summed_nonces[0], &summed_noncesj[0]);
+    secp256k1_ge_set_gej(&summed_nonces[1], &summed_noncesj[1]);
+    if (!secp256k1_musig_compute_noncehash(noncehash, summed_nonces, combined_pk32, msg)) {
         return 0;
     }
+    /* combined_nonce = summed_nonces[0] + b*summed_nonces[1] */
+    secp256k1_scalar_set_b32(&b, noncehash, NULL);
+    secp256k1_ecmult(ecmult_ctx, &combined_noncej, &summed_noncesj[1], &b, NULL);
+    secp256k1_gej_add_ge(&combined_noncej, &combined_noncej, &summed_nonces[0]);
+    secp256k1_ge_set_gej(combined_nonce, &combined_noncej);
+    return 1;
+}
+
+int secp256k1_musig_process_nonces(const secp256k1_context* ctx, secp256k1_musig_session_cache *session_cache, secp256k1_musig_template *sig_template, int *nonce_parity, const unsigned char * const* pubnonces, size_t n_pubnonces, const unsigned char *msg32, const secp256k1_xonly_pubkey *combined_pk, const secp256k1_musig_pre_session *pre_session, const secp256k1_pubkey *adaptor) {
+    secp256k1_gej summed_nonces[2];
+    secp256k1_ge combined_nonce;
+    unsigned char combined_pk32[32];
+    secp256k1_scalar s;
 
+    VERIFY_CHECK(ctx != NULL);
+    ARG_CHECK(secp256k1_ecmult_context_is_built(&ctx->ecmult_ctx));
+    ARG_CHECK(session_cache != NULL);
+    ARG_CHECK(sig_template != NULL);
+    ARG_CHECK(pubnonces != NULL);
+    ARG_CHECK(n_pubnonces > 0);
+    ARG_CHECK(msg32 != NULL);
+    ARG_CHECK(combined_pk != NULL);
+    ARG_CHECK(pre_session != NULL);
+    ARG_CHECK(pre_session->magic == pre_session_magic);
+
+    if (!secp256k1_xonly_pubkey_serialize(ctx, combined_pk32, combined_pk)) {
+        return 0;
+    }
+
+    /* Compute combined nonce and store in sig template */
+    secp256k1_musig_sum_nonces(ctx, summed_nonces, pubnonces, n_pubnonces);
```

Comment:
shouldn't we check whether all the `summed_nonces` are not to infinity as you did in `secp256k1_musig_nonces_combine` ?

Reply:
If any `summed_nonces[i]` is infinity, we can't serialize it and can't serialize and hash it which will result in `musig_process_nonces_internal` failing. This is tested (see `inf_pubnonce_ptr`).